Flaky test: Test/ObservabilityMidStreamFailDeny

```
--- FAIL: Test (1.45s)
    --- FAIL: Test/ObservabilityMidStreamFailDeny (0.07s)
        tlogger.go:133: INFO server.go:734 [core] [Server #378] Server created  (t=+361.095µs)
        tlogger.go:133: INFO server.go:930 [core] [Server #378 ListenSocket #379] ListenSocket created  (t=+452.075µs)
        tlogger.go:133: INFO server.go:734 [core] [Server #380] Server created  (t=+834.4µs)
        stubserver.go:300: Started test service backend at "127.0.0.1:43241"
        setup.go:45: Created new snapshot cache...
        tlogger.go:133: INFO server.go:930 [core] [Server #380 ListenSocket #381] ListenSocket created  (t=+990.101µs)
        tlogger.go:133: INFO server.go:734 [core] [Server #382] Server created  (t=+1.421528ms)
        setup.go:45: Registered Aggregated Discovery Service (ADS)...
        setup.go:45: xDS management server serving at: 127.0.0.1:43651...
        tlogger.go:133: INFO server.go:930 [core] [Server #382 ListenSocket #383] ListenSocket created  (t=+1.579994ms)
        server.go:229: Created new resource snapshot...
        logging.go:30: setting snapshot for node 86814d2c-b629-4d5e-b7ae-ded877e722e7
        server.go:235: Updated snapshot cache with resource snapshot...
        tlogger.go:133: INFO clientconn.go:1823 [core] original dial target is: "xds:///test-service"  (t=+2.414684ms)
        tlogger.go:133: INFO clientconn.go:514 [core] [Channel #384] Channel created for target "xds:///test-service"  (t=+2.436975ms)
        tlogger.go:133: INFO clientconn.go:245 [core] [Channel #384] parsed dial target is: resolver.Target{URL:url.URL{Scheme:"xds", Opaque:"", User:(*url.Userinfo)(nil), Host:"", Path:"/test-service", Fragment:"", RawQuery:"", RawPath:"", RawFragment:"", ForceQuery:false, OmitHost:false}}  (t=+2.457794ms)
        tlogger.go:133: INFO clientconn.go:246 [core] [Channel #384] Channel authority set to "test-service"  (t=+2.472632ms)
        tlogger.go:133: INFO clientconn.go:619 [core] [Channel #384] Channel Connectivity change to CONNECTING  (t=+2.513398ms)
        tlogger.go:133: INFO pool.go:309 [xds] xDS node ID: 86814d2c-b629-4d5e-b7ae-ded877e722e7  (t=+2.72847ms)
        tlogger.go:133: INFO xds_resolver.go:160 [xds] [xds-resolver 0x3547903982c0] Creating resolver for target: xds:///test-service  (t=+2.754649ms)
        tlogger.go:133: INFO clientconn.go:1823 [core] original dial target is: "passthrough:///127.0.0.1:43651"  (t=+2.830521ms)
        tlogger.go:133: INFO clientconn.go:514 [core] [Channel #385] Channel created for target "passthrough:///127.0.0.1:43651"  (t=+2.853023ms)
        tlogger.go:133: INFO clientconn.go:245 [core] [Channel #385] parsed dial target is: resolver.Target{URL:url.URL{Scheme:"passthrough", Opaque:"", User:(*url.Userinfo)(nil), Host:"", Path:"/127.0.0.1:43651", Fragment:"", RawQuery:"", RawPath:"", RawFragment:"", ForceQuery:false, OmitHost:false}}  (t=+2.8728ms)
        tlogger.go:133: INFO clientconn.go:246 [core] [Channel #385] Channel authority set to "127.0.0.1:43651"  (t=+2.88882ms)
        tlogger.go:133: INFO clientconn.go:419 [core] [Channel #384] Channel exiting idle mode  (t=+2.94771ms)
        tlogger.go:133: INFO clientconn.go:619 [core] [Channel #385] Channel Connectivity change to CONNECTING  (t=+2.991312ms)
        tlogger.go:133: INFO balancer_wrapper.go:121 [core] [Channel #385] Channel switches to new LB policy "pick_first"  (t=+3.064268ms)
        tlogger.go:133: INFO clientconn.go:922 [core] [Channel #385 SubChannel #386] Subchannel created  (t=+3.106487ms)
        tlogger.go:133: INFO clientconn.go:419 [core] [Channel #385] Channel exiting idle mode  (t=+3.131914ms)
        tlogger.go:133: INFO clientconn.go:1302 [core] [Channel #385 SubChannel #386] Subchannel Connectivity change to CONNECTING  (t=+3.218426ms)
        tlogger.go:133: INFO clientconn.go:1470 [core] [Channel #385 SubChannel #386] Subchannel picks a new address "127.0.0.1:43651" to connect  (t=+3.235358ms)
        tlogger.go:133: INFO clientconn.go:1302 [core] [Channel #385 SubChannel #386] Subchannel Connectivity change to READY  (t=+3.749449ms)
        tlogger.go:133: INFO clientconn.go:619 [core] [Channel #385] Channel Connectivity change to READY  (t=+3.775117ms)
        logging.go:30: respond type.googleapis.com/envoy.config.listener.v3.Listener (requested [test-service]) version "" with version "1" and resources [test-service]
        logging.go:30: open watch 1 for type.googleapis.com/envoy.config.listener.v3.Listener map[test-service:{}] from nodeID "86814d2c-b629-4d5e-b7ae-ded877e722e7", version "1"
        logging.go:30: respond type.googleapis.com/envoy.config.route.v3.RouteConfiguration (requested [route-test-service]) version "" with version "1" and resources [route-test-service]
        logging.go:30: open watch 2 for type.googleapis.com/envoy.config.route.v3.RouteConfiguration map[route-test-service:{}] from nodeID "86814d2c-b629-4d5e-b7ae-ded877e722e7", version "1"
        logging.go:30: respond type.googleapis.com/envoy.config.cluster.v3.Cluster (requested [cluster-test-service]) version "" with version "1" and resources [cluster-test-service]
        logging.go:30: open watch 3 for type.googleapis.com/envoy.config.cluster.v3.Cluster map[cluster-test-service:{}] from nodeID "86814d2c-b629-4d5e-b7ae-ded877e722e7", version "1"
        logging.go:30: respond type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment (requested [endpoints-test-service]) version "" with version "1" and resources [endpoints-test-service]
        tlogger.go:133: INFO clientconn.go:1823 [core] original dial target is: "127.0.0.1:33913"  (t=+4.958849ms)
        tlogger.go:133: INFO clientconn.go:514 [core] [Channel #389] Channel created for target "127.0.0.1:33913"  (t=+5.020495ms)
        tlogger.go:133: INFO clientconn.go:245 [core] [Channel #389] parsed dial target is: resolver.Target{URL:url.URL{Scheme:"dns", Opaque:"", User:(*url.Userinfo)(nil), Host:"", Path:"/127.0.0.1:33913", Fragment:"", RawQuery:"", RawPath:"", RawFragment:"", ForceQuery:false, OmitHost:false}}  (t=+5.08214ms)
        tlogger.go:133: INFO clientconn.go:246 [core] [Channel #389] Channel authority set to "127.0.0.1:33913"  (t=+5.108279ms)
        tlogger.go:133: INFO balancer_wrapper.go:121 [core] [Channel #384] Channel switches to new LB policy "xds_cluster_manager_experimental"  (t=+5.197555ms)
        tlogger.go:133: INFO clustermanager.go:58 [xds] [xds-cluster-manager-lb 0x3547905a7760] Created  (t=+5.222693ms)
        tlogger.go:133: INFO balancergroup.go:274 [xds] [xds-cluster-manager-lb 0x3547905a7760] Adding child policy of type "cds_experimental" for child "cluster:cluster-test-service"  (t=+5.238342ms)
        tlogger.go:133: INFO balancergroup.go:102 [xds] [xds-cluster-manager-lb 0x3547905a7760] Creating child policy of type "cds_experimental" for child "cluster:cluster-test-service"  (t=+5.252117ms)
        tlogger.go:133: INFO cdsbalancer.go:90 [xds] [cds-lb 0x35479045c1c0] Created  (t=+5.267316ms)
        tlogger.go:133: INFO cdsbalancer.go:167 [xds] [cds-lb 0x35479045c1c0] Received balancer config update: {
              "LoadBalancingConfig": null,
              "cluster": "cluster-test-service",
              "isDynamic": false
            }  (t=+5.287504ms)
        tlogger.go:133: INFO balancer.go:76 [xds] [priority-lb 0x3547900d8cb0] Created  (t=+5.362774ms)
        tlogger.go:133: INFO balancergroup.go:528 [xds] [xds-cluster-manager-lb 0x3547905a7760] Balancer state update from child cluster:cluster-test-service, new state: {ConnectivityState:CONNECTING Picker:0x354790412200}  (t=+5.575843ms)
        tlogger.go:133: INFO balancerstateaggregator.go:147 [xds] [xds-cluster-manager-lb 0x3547905a7760] State update from sub-balancer "cluster:cluster-test-service": {ConnectivityState:CONNECTING Picker:0x354790412200}  (t=+5.606239ms)
        tlogger.go:133: INFO balancergroup.go:274 [xds] [priority-lb 0x3547900d8cb0] Adding child policy of type "outlier_detection_experimental" for child "priority-0-0"  (t=+5.629313ms)
        tlogger.go:133: INFO balancergroup.go:102 [xds] [priority-lb 0x3547900d8cb0] Creating child policy of type "outlier_detection_experimental" for child "priority-0-0"  (t=+5.644281ms)
        tlogger.go:133: INFO balancer.go:94 [xds] [outlier-detection-lb 0x354790463950] Created  (t=+5.663526ms)
        tlogger.go:133: INFO clusterimpl.go:89 [xds] [xds-cluster-impl-lb 0x3547904b7320] Created  (t=+5.684295ms)
        tlogger.go:133: INFO clusterimpl.go:101 [xds] [xds-cluster-impl-lb 0x3547904b7320] xDS credentials in use: false  (t=+5.699874ms)
        tlogger.go:133: INFO weightedtarget.go:65 [xds] [weighted-target-lb 0x3547905a7bc0] Created  (t=+5.768643ms)
        tlogger.go:133: INFO balancer.go:88 [xds] [wrrlocality-lb 0x3547900f9b90] Created  (t=+5.789091ms)
        tlogger.go:133: INFO balancergroup.go:274 [xds] [weighted-target-lb 0x3547905a7bc0] Adding child policy of type "round_robin" for child "{region=\"region-1\", zone=\"zone-1\", sub_zone=\"subzone-1\"}"  (t=+5.828966ms)
        tlogger.go:133: INFO balancergroup.go:102 [xds] [weighted-target-lb 0x3547905a7bc0] Creating child policy of type "round_robin" for child "{region=\"region-1\", zone=\"zone-1\", sub_zone=\"subzone-1\"}"  (t=+5.84748ms)
        tlogger.go:133: INFO roundrobin.go:56 [roundrobin] [0x35479098b080] Created  (t=+5.866336ms)
        tlogger.go:133: INFO clientconn.go:922 [core] [Channel #384 SubChannel #390] Subchannel created  (t=+5.906791ms)
        tlogger.go:133: INFO balancergroup.go:528 [xds] [weighted-target-lb 0x3547905a7bc0] Balancer state update from child {region="region-1", zone="zone-1", sub_zone="subzone-1"}, new state: {ConnectivityState:CONNECTING Picker:0x3547908f9b40}  (t=+5.94898ms)
        tlogger.go:133: INFO aggregator.go:253 [xds] [weighted-target-lb 0x3547905a7bc0] Child pickers with config: map[{region="region-1", zone="zone-1", sub_zone="subzone-1"}:weight:1,picker:0x3547908f9b40,state:CONNECTING,stateToAggregate:CONNECTING]  (t=+5.971011ms)
        tlogger.go:133: INFO clientconn.go:1302 [core] [Channel #384 SubChannel #390] Subchannel Connectivity change to CONNECTING  (t=+6.007389ms)
        tlogger.go:133: INFO clientconn.go:1470 [core] [Channel #384 SubChannel #390] Subchannel picks a new address "localhost:43241" to connect  (t=+6.016747ms)
        tlogger.go:133: INFO balancergroup.go:528 [xds] [priority-lb 0x3547900d8cb0] Balancer state update from child priority-0-0, new state: {ConnectivityState:CONNECTING Picker:0x3547906b59b0}  (t=+6.175423ms)
        tlogger.go:133: INFO balancergroup.go:528 [xds] [xds-cluster-manager-lb 0x3547905a7760] Balancer state update from child cluster:cluster-test-service, new state: {ConnectivityState:CONNECTING Picker:0x3547906b59b0}  (t=+6.187115ms)
        tlogger.go:133: INFO balancerstateaggregator.go:147 [xds] [xds-cluster-manager-lb 0x3547905a7760] State update from sub-balancer "cluster:cluster-test-service": {ConnectivityState:CONNECTING Picker:0x3547906b59b0}  (t=+6.195401ms)
        tlogger.go:133: INFO balancerstateaggregator.go:191 [xds] [xds-cluster-manager-lb 0x3547905a7760] Child pickers: map[cluster:cluster-test-service:picker:0x3547906b59b0,state:CONNECTING,stateToAggregate:CONNECTING]  (t=+6.206401ms)
        tlogger.go:133: INFO clientconn.go:619 [core] [Channel #389] Channel Connectivity change to CONNECTING  (t=+6.241106ms)
        tlogger.go:133: INFO balancer_wrapper.go:121 [core] [Channel #389] Channel switches to new LB policy "pick_first"  (t=+6.264019ms)
        tlogger.go:133: INFO clientconn.go:922 [core] [Channel #389 SubChannel #391] Subchannel created  (t=+6.290438ms)
        tlogger.go:133: INFO clientconn.go:419 [core] [Channel #389] Channel exiting idle mode  (t=+6.302942ms)
        tlogger.go:133: INFO clientconn.go:1302 [core] [Channel #389 SubChannel #391] Subchannel Connectivity change to CONNECTING  (t=+6.331425ms)
        tlogger.go:133: INFO clientconn.go:1470 [core] [Channel #389 SubChannel #391] Subchannel picks a new address "127.0.0.1:33913" to connect  (t=+6.340512ms)
        logging.go:30: open watch 4 for type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment map[endpoints-test-service:{}] from nodeID "86814d2c-b629-4d5e-b7ae-ded877e722e7", version "1"
        tlogger.go:133: INFO clientconn.go:1302 [core] [Channel #384 SubChannel #390] Subchannel Connectivity change to READY  (t=+8.244501ms)
        tlogger.go:133: INFO balancergroup.go:528 [xds] [weighted-target-lb 0x3547905a7bc0] Balancer state update from child {region="region-1", zone="zone-1", sub_zone="subzone-1"}, new state: {ConnectivityState:READY Picker:0x3547903d3280}  (t=+8.284796ms)
        tlogger.go:133: INFO aggregator.go:253 [xds] [weighted-target-lb 0x3547905a7bc0] Child pickers with config: map[{region="region-1", zone="zone-1", sub_zone="subzone-1"}:weight:1,picker:0x3547903d3280,state:READY,stateToAggregate:READY]  (t=+8.305665ms)
        tlogger.go:133: INFO balancergroup.go:528 [xds] [priority-lb 0x3547900d8cb0] Balancer state update from child priority-0-0, new state: {ConnectivityState:READY Picker:0x3547907011a0}  (t=+8.328819ms)
        tlogger.go:133: INFO balancergroup.go:528 [xds] [xds-cluster-manager-lb 0x3547905a7760] Balancer state update from child cluster:cluster-test-service, new state: {ConnectivityState:READY Picker:0x3547907011a0}  (t=+8.354937ms)
        tlogger.go:133: INFO balancerstateaggregator.go:147 [xds] [xds-cluster-manager-lb 0x3547905a7760] State update from sub-balancer "cluster:cluster-test-service": {ConnectivityState:READY Picker:0x3547907011a0}  (t=+8.369885ms)
        tlogger.go:133: INFO balancerstateaggregator.go:191 [xds] [xds-cluster-manager-lb 0x3547905a7760] Child pickers: map[cluster:cluster-test-service:picker:0x3547907011a0,state:READY,stateToAggregate:READY]  (t=+8.384022ms)
        tlogger.go:133: INFO clientconn.go:619 [core] [Channel #384] Channel Connectivity change to READY  (t=+8.401705ms)
        tlogger.go:133: INFO clientconn.go:1302 [core] [Channel #389 SubChannel #391] Subchannel Connectivity change to READY  (t=+8.460975ms)
        tlogger.go:133: INFO clientconn.go:619 [core] [Channel #389] Channel Connectivity change to READY  (t=+8.483558ms)
        ext_proc_ext_test.go:4153: Expected error from processor stream failure, got none after 10 messages
        tlogger.go:133: INFO clientconn.go:619 [core] [Channel #384] Channel Connectivity change to SHUTDOWN  (t=+9.503975ms)
        tlogger.go:133: INFO resolver_wrapper.go:112 [core] [Channel #384] Closing the name resolver  (t=+9.523241ms)
        tlogger.go:133: INFO balancer_wrapper.go:159 [core] [Channel #384] ccBalancerWrapper: closing  (t=+9.541125ms)
        tlogger.go:133: INFO clusterimpl.go:523 [xds] [xds-cluster-impl-lb 0x3547904b7320] Shutdown  (t=+9.682218ms)
        tlogger.go:133: INFO balancergroup.go:351 [xds] [priority-lb 0x3547900d8cb0] Removing child policy for child "priority-0-0"  (t=+9.70439ms)
        tlogger.go:133: INFO cdsbalancer.go:416 [xds] [cds-lb 0x35479045c1c0] Shutdown  (t=+9.742681ms)
        tlogger.go:133: INFO clustermanager.go:192 [xds] [xds-cluster-manager-lb 0x3547905a7760] Shutdown  (t=+9.755916ms)
        tlogger.go:129: WARNING ads_stream.go:410 [xds] [xds-client 0x3547900e8fa0] [xds-channel 0x35478ffb4870] [ads-stream 0x3547900e9040] Sending ADS request for type "type.googleapis.com/envoy.config.listener.v3.Listener", resources: [], version: "1", nonce: "1" failed: EOF  (t=+9.780432ms)
        tlogger.go:129: WARNING ads_stream.go:452 [xds] [xds-client 0x3547900e8fa0] [xds-channel 0x35478ffb4870] [ads-stream 0x3547900e9040] ADS stream closed: rpc error: code = Canceled desc = context canceled  (t=+9.799447ms)
        tlogger.go:133: INFO ads_stream.go:165 [xds] [xds-client 0x3547900e8fa0] [xds-channel 0x35478ffb4870] [ads-stream 0x3547900e9040] Shutdown ADS stream  (t=+9.82736ms)
        tlogger.go:133: INFO clientconn.go:619 [core] [Channel #385] Channel Connectivity change to SHUTDOWN  (t=+9.844101ms)
        tlogger.go:133: INFO resolver_wrapper.go:112 [core] [Channel #385] Closing the name resolver  (t=+9.858097ms)
        tlogger.go:133: INFO balancer_wrapper.go:159 [core] [Channel #385] ccBalancerWrapper: closing  (t=+9.873306ms)
        tlogger.go:133: INFO clientconn.go:1302 [core] [Channel #385 SubChannel #386] Subchannel Connectivity change to SHUTDOWN  (t=+9.909523ms)
        tlogger.go:133: INFO clientconn.go:1696 [core] [Channel #385 SubChannel #386] Subchannel deleted  (t=+9.922838ms)
        tlogger.go:133: INFO clientconn.go:1231 [core] [Channel #385] Channel deleted  (t=+10.002527ms)
        tlogger.go:133: INFO channel.go:143 [xds] [xds-client 0x3547900e8fa0] [xds-channel 0x35478ffb4870] Shutdown  (t=+10.021532ms)
        tlogger.go:133: INFO xdsclient.go:210 [xds] [xds-client 0x3547900e8fa0] Shutdown  (t=+10.044555ms)
        tlogger.go:133: INFO clientconn.go:619 [core] [Channel #389] Channel Connectivity change to SHUTDOWN  (t=+10.060886ms)
        tlogger.go:133: INFO resolver_wrapper.go:112 [core] [Channel #389] Closing the name resolver  (t=+10.086023ms)
        tlogger.go:133: INFO balancer_wrapper.go:159 [core] [Channel #389] ccBalancerWrapper: closing  (t=+10.10033ms)
        tlogger.go:133: INFO clientconn.go:1302 [core] [Channel #389 SubChannel #391] Subchannel Connectivity change to SHUTDOWN  (t=+10.133742ms)
        tlogger.go:133: INFO clientconn.go:1696 [core] [Channel #389 SubChannel #391] Subchannel deleted  (t=+10.153189ms)
        tlogger.go:133: INFO clientconn.go:1231 [core] [Channel #389] Channel deleted  (t=+11.593451ms)
        tlogger.go:133: INFO xds_resolver.go:301 [xds] [xds-resolver 0x3547903982c0] Shutdown  (t=+11.620321ms)
        tlogger.go:133: INFO clientconn.go:1302 [core] [Channel #384 SubChannel #390] Subchannel Connectivity change to SHUTDOWN  (t=+11.654966ms)
        tlogger.go:133: INFO clientconn.go:1696 [core] [Channel #384 SubChannel #390] Subchannel deleted  (t=+11.6732ms)
        tlogger.go:133: INFO clientconn.go:1231 [core] [Channel #384] Channel deleted  (t=+11.779138ms)
        tlogger.go:133: INFO server.go:866 [core] [Server #380 ListenSocket #381] ListenSocket deleted  (t=+11.827839ms)
        tlogger.go:133: INFO server.go:866 [core] [Server #382 ListenSocket #383] ListenSocket deleted  (t=+11.910494ms)
        tlogger.go:133: INFO server.go:866 [core] [Server #378 ListenSocket #379] ListenSocket deleted  (t=+11.955367ms)
FAIL

```

--- a/internal/xds/httpfilter/extproc/ext_proc_ext_test.go
+++ b/internal/xds/httpfilter/extproc/ext_proc_ext_test.go
@@ -4135,12 +4135,15 @@ func (s) TestObservabilityMidStreamFailDeny(t *testing.T) {
 		t.Fatalf("FullDuplexCall() failed: %v", err)
 	}
 
-	const numMessages = 10
 	var gotErr error
-	// Send and recv 10 messages and verify that the RPC fails with correct error.
-	for i := 0; i < numMessages; i++ {
+	// Continuously send and recv messages until the RPC fails with correct error
+	// or the test context times out.
+	for {
+		if ctx.Err() != nil {
+			t.Fatalf("Timed out waiting for stream to fail from processor disconnect: %v", ctx.Err())
+		}
 		req := &testpb.StreamingOutputCallRequest{
-			Payload: &testpb.Payload{Body: []byte(fmt.Sprintf("msg-%d", i))},
+			Payload: &testpb.Payload{Body: []byte("msg")},
 		}
 		if err := stream.Send(req); err != nil {
 			gotErr = err
@@ -4152,8 +4155,11 @@ func (s) TestObservabilityMidStreamFailDeny(t *testing.T) {
 		}
 	}
 
+	if ctx.Err() != nil {
+		t.Fatalf("Timed out waiting for stream to fail from processor disconnect: %v", ctx.Err())
+	}
 	if gotErr == nil {
-		t.Fatalf("Expected error from processor stream failure, got none after %d messages", numMessages)
+		t.Fatalf("Expected error from processor stream failure, got none")
 	}
 	if code := status.Code(gotErr); code != codes.Internal {
 		t.Fatalf("Stream returned status code %v (error: %v), want %v", code, gotErr, codes.Internal)

